resources/packs/k8s/k8s.lr

```diff
@@ -622,6 +622,8 @@ k8s.ingress @defaults("namespace name created") {
   manifest dict
   // Ingress rules
   rules []k8s.ingressrule
+  // Certificates referenced by Ingress
+  certificates []core.certificate
```

Comment:
I think we might need a bit more than just a list of certificates here. If I look at the `Ingress` documentation you specify a `tls` section that can have a list of hosts and for each list of hosts you have certificates.

```yaml
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  name: tls-example-ingress
spec:
  tls:
  - hosts:
      - https-example.foo.com
    secretName: testsecret-tls
...
I can imagine that if you are writing a control for an ingress you might want to locate a specific certificate for a specific set of hosts, which isn't possible with the current resource
```

Reply:
Good catch. No clue what I was thinking by just dumping all the certificates found in each referenced Secret into a single list of []certificate. I've added a new structure to more closely capture the way the data is structured in the Ingress resource.